### ISSUE TITLE
[IOS-XE] "show ip pim neighbor" does not seem to support LISP interfaces

### ISSUE THREAD

**Issue by desiredrive** [CiscoTestAutomation/genieparser#901]
When parsing PIM neighbors with the following output, the parser can do it right:

```
show ip pim neighbor
PIM Neighbor Table
Mode: B - Bidir Capable, DR - Designated Router, N - Default DR Priority,
      P - Proxy Capable, S - State Refresh Capable, G - GenID Capable,
      L - DR Load-balancing Capable
Neighbor          Interface                Uptime/Expires    Ver   DR
Address                                                            Prio/Mode
172.19.2.6        TenGigabitEthernet1/0/3  4d23h/00:01:41    v2    1 / S P G
172.19.2.8        TenGigabitEthernet1/0/7  4d23h/00:01:21    v2    1 / S P G
```

But if a LISP interface exists, the parser fails:
```
show ip pim vrf Campus neighbor
PIM Neighbor Table
Mode: B - Bidir Capable, DR - Designated Router, N - Default DR Priority,
      P - Proxy Capable, S - State Refresh Capable, G - GenID Capable,
      L - DR Load-balancing Capable
Neighbor          Interface                Uptime/Expires    Ver   DR
Address                                                            Prio/Mode
172.19.1.64       LISP0.4099               4d23h/00:01:11    v2    0 /
```

Output of the first example:

`{'vrf': {'default': {'interfaces': {'TenGigabitEthernet1/0/3': {'address_family': {'ipv4': {'neighbors': {'172.19.2.6': {'expiration': '00:01:41', 'dr_priority': 1, 'up_time': '4d23h', 'interface': 'TenGigabitEthernet1/0/3', 'version': 'v2', 'state_refresh_capable': True, 'proxy_capable': True, 'genid_capable': True}}}}}, 'TenGigabitEthernet1/0/7': {'address_family': {'ipv4': {'neighbors': {'172.19.2.8': {'expiration': '00:01:21', 'dr_priority': 1, 'up_time': '4d23h', 'interface': 'TenGigabitEthernet1/0/7', 'version': 'v2', 'state_refresh_capable': True, 'proxy_capable': True, 'genid_capable': True}}}}}}}}, '_exclude': ['expiration', 'up_time']}
`

For the second.

```
None
Genie exception****
```

Observed on the IOS-XE parser, C9300 in 17.12.2

**Comment by SohanTirpude** [comment]
Hello @desiredrive,

For this, I was able to find the issue and possible solution as well. With the fix, the output looks like this:
```
{
    "vrf": {
        "Campus": {
            "interfaces": {
                "LISP0.4099": {
                    "address_family": {
                        "ipv4": {
                            "neighbors": {
                                "172.19.1.64": {
                                    "expiration": "00:01:11",
                                    "dr_priority": 0,
                                    "up_time": "4d23h",
                                    "interface": "LISP0.4099",
                                    "version": "v2"
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```
Kindly check and confirm so I can raise PR and get it merged soon.

Thank you.

**Comment by desiredrive** [comment]
Hi Sohan, it looks good, I'd say it is what I expected. Thanks!

**Comment by SohanTirpude** [comment]
Hello @desiredrive,

Thank you for the confirmation. I have raised the PR and once it is merged, I will update you back here.

Thank you.

**Comment by SohanTirpude** [comment]
Hello @desiredrive,

The PR got merged and the solution will be available in the upcoming pyats 24.10 version. Hence, I am closing this ticket for now. If you are facing any other issues, kindly open a new ticket.

Thank you.
